scipy/ndimage/tests/test_filters.py

```diff
@@ -2909,44 +2936,46 @@ def test_input_validation(self):
         with pytest.raises(ValueError, match=message):
             ndimage.vectorized_filter(input, function, size=size, mode='valid', cval=1)
 
-        message = "`cval` must include only numbers."
-        with pytest.raises(ValueError, match=message):
+        other_messages = "|Unsupported|The array_api_strict|new|Value 'a duck'"
+        message = "`cval` must include only numbers." + other_messages
+        with pytest.raises((ValueError, TypeError), match=message):
             ndimage.vectorized_filter(input, function, size=size,
-                                      mode='constant', cval='a duck')
+                              mode='constant', cval='a duck')
 
-        message = "`batch_memory` must be positive number."
+        message = "`batch_memory` must be positive number." + other_messages
         with pytest.raises(ValueError, match=message):
             ndimage.vectorized_filter(input, function, size=size, batch_memory=0)
         with pytest.raises(ValueError, match=message):
             ndimage.vectorized_filter(input, function, size=size, batch_memory=(1, 2))
-        with pytest.raises(ValueError, match=message):
-            ndimage.vectorized_filter(input, function, size=size,
-                                      batch_memory="shrubbery")
+        with pytest.raises((ValueError, TypeError), match=message):
+            ndimage.vectorized_filter(input, function, size=size, batch_memory="a duck")
 
+    @pytest.mark.skip_xp_backends('torch', reason="Need to debug.")
```

Comment:
What to debug it here or kick the can down the road?

Reply:
I'll take a quick look this afternoon and at least provide a more informative reason. 

I didn't remember that there were other appearances of "xp.pad"? If others remain, I'll clarify that it's NumPy or CuPy.

Reply:
Fixed. Just a PyTorch `size` gotcha.